builtin/id_hash.h

```diff
@@ -203,6 +203,53 @@ struct IdHashes {
         }
         return true;
     }
+
+    struct const_iterator
```

Comment:
I find boost::iterator_facade very helpful for getting this kind of iterator right.